src/only_child.rs

```diff
@@ -0,0 +1,203 @@
+    /// [`Result`]: https://doc.rust-lang.org/std/result/enum.Result.html
+    fn try_only_child_with_filter<'a, P>(&'a self, filter: P) -> Result<&'a Self, Error>
+    where
+        P: Fn(&'a Self) -> bool;
+
+    /// Get the unique child of an element.
+    ///
+    /// To select this element, a predicate is specified taking the element as
+    /// an input and returning a boolean.
+    ///
+    /// The function returns an [`Option`] with the child if there is one and
+    /// only one child corresponding to the predicate.
+    ///
+    /// [`Option`]:  https://doc.rust-lang.org/std/option/enum.Option.html
+    fn only_child_with_filter<'a, P>(&'a self, filter: P) -> Option<&'a Self>
+    where
+        P: Fn(&'a Self) -> bool,
+    {
+        self.try_only_child_with_filter(filter).ok()
+    }
+
+    /// Try to get an unique child from its name and return a [`Result`].
+    ///
+    /// Returns an [`Error`] if the child can't be found ([`NoChild`])
+    /// or if the child is not unique ([`NotUniqueChild`])
+    ///
+    /// [`Error`]: enum.Error.html
+    /// [`NoChild`]: enum.Error.html#variant.NoChild
+    /// [`NotUniqueChild`]: enum.Error.html#variant.NotUniqueChild
+    /// [`Result`]: https://doc.rust-lang.org/std/result/enum.Result.html
+    fn try_only_child<'a>(&'a self, child_name: &str) -> Result<&'a Self, Error>;
+
+    /// Get a unique child from its name and return an [`Option`].
+    ///
+    /// Returns [`None`] if the child can't be found or if the child is not
+    /// unique.
+    ///
+    /// [`None`]:  https://doc.rust-lang.org/std/option/enum.Option.html#variant.None
+    /// [`Option`]:  https://doc.rust-lang.org/std/option/enum.Option.html
+    fn only_child<'a>(&'a self, child_name: &str) -> Option<&'a Self> {
+        self.try_only_child(child_name).ok()
+    }
+}
+
+impl OnlyChildElementExt for Element {
+    /// Implementation of [`OnlyChildElementExt`] for [`Element`] gives you the ability to
+    /// select one and only one child of an XML tag depending on a predicate. If
+    /// none or more than two are found with the predicate, an error is
+    /// returned.
+    ///
+    /// ```
+    /// use minidom::Element;
+    /// use minidom_ext::OnlyChildElementExt;
+    ///
+    /// let xml: &'static str = r#"<root>
+    ///         <child type="ugly" />
+    ///         <child />
+    ///     </root>"#;
+    /// let root: Element = xml.parse().unwrap();
+    /// let child = root
+    ///     .try_only_child_with_filter(|e| {
+    ///         e.name() == "child" && e.attr("type").map(|id| id == "ugly").unwrap_or(false)
+    ///     })
+    ///     .unwrap();
+    /// assert_eq!("child", child.name());
+    /// ```
+    ///
+    /// [`OnlyChildElementExt`]: trait.OnlyChildElementExt.html
+    /// [`Element`]: ../minidom/element/struct.Element.html
+    fn try_only_child_with_filter<'a, P>(&'a self, filter: P) -> Result<&'a Self, Error>
```

Comment:
Not sure at all: maybe rename to have all matching the same (multiple occurences):
```suggestion
    fn try_only_child_with_criterion<'a, P>(&'a self, criterion: P) -> Result<&'a Self, Error>
```

Reply:
See https://github.com/CanalTP/minidom_ext/pull/1#discussion_r409524301.